### ISSUE TITLE
Mathjax fonts are loaded only if I am online

### ISSUE THREAD

**Issue by london111** [dubiousjim/pandoc-iki#2]
I have a local ikiwiki which uses pandoc for displaying math via mathjax. However mathjax ist displayed only when I have a connection to the internet, it seems to load the fonts from the web. 

Is there a way to enable mathjax locally, such that math is displayed when viewing the wiki locally on my laptop, too?

**Comment by dubiousjim** [comment]
On Saturday, November 26, 2011 3:28 AM, "london111"
reply@reply.github.com
wrote:

> I have a local ikiwiki which uses pandoc for displaying math via mathjax.
> However mathjax ist displayed only when I have a connection to the
> internet, it seems to load the fonts from the web. 
> 
> Is there a way to enable mathjax locally, such that math is displayed
> when viewing the wiki locally on my laptop, too?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/dubiousjim/pandoc-iki/issues/2

Hi London111, this issue is not specific to Mathjax or to your ikiwiki
installation. More generally, you need to be able to execute cgi scripts
from your local machine. There are ways to do that without an internet
connection, yes. Most of the ways I remember require that you still run
a webserver on your local machine, and configure it to run cgis at
http://localhost/your/mathjax/path. I've also seen ways to do this
without running a full-fledged webserver. But this is all highly
dependent on what kind of OS and software setup you're working with. I
suggest googling "local cgi <your-favorite-webserver>" and reading from
there. If you get to the point where you can run a simple test cgi
without an internet connection, it should be a small step from there to

## being able to run mathjax in the same way.

dubiousjim@gmail.com

**Comment by london111** [comment]
I am running an apache2 webserver on my laptop locally (os: linux). If I open the html source of one of my wiki-pages, there are the following lines:

```
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML-full" type="text/javascript"></script>
  </head>
  <body>
  <noscript>
  <div style="color:#CC0000; text-align:center">
  <b>Warning: <a href="http://www.mathjax.org/">MathJax</a>
  requires JavaScript to process the mathematics on this page.<br>
  If your browser supports JavaScript, be sure it is enabled.</b>
  </div>
  <hr>
  </noscript>
```

So you see, it points to http://cdn.mathjax.org/mathjax/latest/MathJax.js.

I tried to change the line 

 `push @args, '--mathjax=/dev/null';`

to

`push @args, '--mathjax=/mathjax/MathJax.js';`

in my pandoc.pm

also to 

`push @args, '--mathjax=/var/www/mathjax/MathJax.js';`

But it doesn't help.

**Comment by dubiousjim** [comment]
OK, this is off the top of my head; I haven't worked with these files
recently. But I expect you'll get things sorted out more quickly if I
try to push you in the following direction, even if I can only do so
vaguely and clumsily, than you will by waiting until I'm able to look
into it more closely.

I think this line:

>  push @args, '--mathjax=/dev/null';

is supplying an argument to pandoc. The ikiwiki/pandoc integration
involves pandoc only creating the html fragment that makes up the body
of your page. We tell pandoc not to bother creating the page headers and
footers. Ikiwiki handles that with its templates. So for pandoc I think
all we need is for the --mathjax argument to be there, and I think it
required a path, so I just used /dev/null. I suspect that you could
substitute any path here, and you'd get the same output from pandoc. The
actual content of the path would have only made a difference to the html
header, which we're suppressing.

The link to the mathjax.org site is in the templates you're using. Part
of the install instructions for pandoc-iki says:

> You can use MathJax http://www.mathjax.org/. This requires copying the `mathjax.tmpl` file that accompanies this plugin to a template directory seen by your wiki. Rename the file to `page.tmpl`. (Available since Pandoc 1.8.)

Go find that file, and change the 

> <script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML-full" type="text/javascript"></script>
> 
> inside it to point to your local mathjax install.

Probably I was wrong before. If you make this change, I expect it will
work even without a webserver running on your local machine.

On Monday, November 28, 2011 8:01 AM, "london111"
reply@reply.github.com
wrote:

**Comment by london111** [comment]
Ok, thanks, I just changed the line in  page.tmpl you mentioned such that mathjax points to a local directory (relative to localhost). To make sure that the solution works in firefox too, this directory (or a symbolic link to it) has to be in the wiki destdir (compare http://www.mathjax.org/resources/faqs/)
